src/main/scala/flow.scala

```diff
@@ -8,9 +8,36 @@ import scala.language.experimental.macros
 @compileTimeOnly("implementation detail of flow comprehensions. don't use yourself")
 final class FlowContext private()
 
+abstract class Constructor[H[_]]{
+  def create[T](v: T): H[T]
+}
```

Comment:
I am uncertain about the default .apply macro. I like the incentive, but would like to see instances first where this would work outside of the std lib.